src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,148 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int CAPACITY_MULTIPLIER = 2;
+
+    private Node<K, V>[] table;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        int position = getPositionInTable(key);
+        Node<K, V> newNode = new Node<>(getKeyHashCode(key), key, value, null);
+
+        if (table[position] == null) {
+            table[position] = newNode;
+
+            size++;
+        } else {
+            Node<K, V> currElem = getElementByKeyOrLast(key, position);
 
+            if ((key == null && currElem.getKey() == null)
+                            || (currElem.getKey() != null && currElem.getKey().equals(key))
```

Comment:
Don't complicate this conditional redundant brackets 
```suggestion
            if (key == currElem.getKey()
                 || currElem.getKey() != null && currElem.getKey().equals(key)
```